services/app-api/src/postgres/contractAndRates/updateDraftContract.test.ts

```diff
@@ -40,10 +41,185 @@ describe('updateDraftContract', () => {
         )
     })
 
-    it.todo(
-        'updates linked document tables as expected (via upsert, not create)'
-    )
-    it.todo('updates linked contact table as expected (via upsert, not create)')
+    it('updates linked documents as expected in multiple requests', async () => {
```

Comment:
<strike>Let's add a test like `deletes linked documents and contacts without breaking old revisions.`</strike> Unlock contract is not fully implemented so I don't think we could test this anyways.

Reply:
@JasonLin0991 I will add a todo for this its a good test to have later. I don't think this solution runs that risk because udpate only happens on new drafts and it delete things only related to that specific

Reply:
> Also, in the Prisma schema, documents and revisions have a one-to-many relationship, so a document could not be related to many contract revisions. So new revisions just create a whole new set of documents?

You are correct with the way we set up the relationship, a document can never be related to more than one revision. So deleting a document that can only be related to one revision would not encounter an error where there the document has another relationship, and it would either cascade the delete to other related records or return an error because there were other relationships.

There could be a discussion on whether creating new records on unlock is something we want to continue. Since technically they are not "new" documents.

Reply:
@JasonLin0991 Interesting, tis feels like an architectural discussion that goes beyond even the scope of the issues you are raising about how updating resources should work elsewhere in this PR,- the premise of our audit log has been that we create a distinct and new record even in protos. I have been assuming we would copy over all of our scalar fields similarly with contract and rates revisions and that we would also copy over the documents/contracts.